Build of 2.7 branch fails when MBEDTLS_CERTS_C is disabled

### Description
- Type: Bug (compilation error)
- Priority: Minor blocker (exotic ;p)

---------------------------------------------------------------
## Bug

**OS**  
Any

**mbed TLS build:**  
Version: 2.7.x only, from 2.7.11 on

Configuration: default with `MBEDTLS_CERTS_C` disabled
Compiler: Tested with GCC on Linux, Clang on macOS and the Android NDK's Clang

**Analysis**  

This only happens with the 2.7 LTS versions. Not affected: 2.16.6 LTS, 2.25.0, `development` and `master`, as far as I tested.

Reason: Caused by wrongfully placed closing brackets between `#ifdef`'s, copy & paste hell unleashed ;-)

Origin: The bug was introduced in PR #2498

**Steps to reproduce**  

Run (Linux example):
1. `scripts/config.pl unset MBEDTLS_CERTS_C`
2. (optional:) `sed -e "s|DESTDIR=/usr/local|DESTDIR=$(pwd)/build|" -i Makefile`
3. `make lib`
4. `make install`

Note: If you only run `make` in step 3, the compiler errors shown below are easily overlooked due to the scrolling.

**Remarks**  

I've already fixed the bug locally and will provide a PR soon.

Question: Is the actual build of the `programs` folder intended, in the `install` stage, when just `make libs` was run?

**Compiler log**  
```
+ make install
  CC    aes/aescrypt2.c
  CC    aes/crypt_and_hash.c
  CC    hash/hello.c
  CC    hash/generic_sum.c
  CC    pkey/dh_client.c
  CC    pkey/dh_genprime.c
  CC    pkey/dh_server.c
  CC    pkey/ecdh_curve25519.c
  CC    pkey/ecdsa.c
  CC    pkey/gen_key.c
  CC    pkey/key_app.c
  CC    pkey/key_app_writer.c
  CC    pkey/mpi_demo.c
  CC    pkey/pk_decrypt.c
  CC    pkey/pk_encrypt.c
  CC    pkey/pk_sign.c
  CC    pkey/pk_verify.c
  CC    pkey/rsa_genkey.c
  CC    pkey/rsa_decrypt.c
  CC    pkey/rsa_encrypt.c
  CC    pkey/rsa_sign.c
  CC    pkey/rsa_verify.c
  CC    pkey/rsa_sign_pss.c
  CC    pkey/rsa_verify_pss.c
  CC    ssl/dtls_client.c
  CC    ssl/dtls_server.c
  CC    ssl/ssl_client1.c
  CC    ssl/ssl_client2.c
ssl/ssl_client2.c:1904:49: error: expected '}'
          MBEDTLS_CTR_DRBG_C MBEDTLS_TIMING_C */
                                                ^
ssl/ssl_client2.c:1179:5: note: to match this '{'
    {
    ^
ssl/ssl_client2.c:1904:49: error: expected '}'
          MBEDTLS_CTR_DRBG_C MBEDTLS_TIMING_C */
                                                ^
ssl/ssl_client2.c:1142:5: note: to match this '{'
    {
    ^
ssl/ssl_client2.c:1904:49: error: expected '}'
          MBEDTLS_CTR_DRBG_C MBEDTLS_TIMING_C */
                                                ^
ssl/ssl_client2.c:478:1: note: to match this '{'
{
^
3 errors generated.
make[1]: *** [ssl/ssl_client2] Error 1
make: *** [programs] Error 2
```

--- a/ChangeLog.d/build-without-certs.txt
+++ b/ChangeLog.d/build-without-certs.txt
@@ -1,0 +1,3 @@
+Bugfix
+   * Fix the build of sample programs when MBEDTLS_PEM_C is enabled but
+     MBEDTLS_CERTS_C is disabled. Reported by Michael Schuster in #4206.

--- a/programs/ssl/ssl_client2.c
+++ b/programs/ssl/ssl_client2.c
@@ -1145,8 +1145,8 @@ int main( int argc, char *argv[] )
         mbedtls_printf( "MBEDTLS_CERTS_C not defined." );
 #else
         mbedtls_printf( "All test CRTs loaded via MBEDTLS_CERTS_C are PEM-encoded, but MBEDTLS_PEM_PARSE_C is disabled." );
-    }
 #endif /* MBEDTLS_CERTS_C */
+    }
 #endif /* MBEDTLS_CERTS_C && MBEDTLS_PEM_PARSE_C */
     if( ret < 0 )
     {
@@ -1182,8 +1182,8 @@ int main( int argc, char *argv[] )
         mbedtls_printf( "MBEDTLS_CERTS_C not defined." );
 #else
         mbedtls_printf( "All test CRTs loaded via MBEDTLS_CERTS_C are PEM-encoded, but MBEDTLS_PEM_PARSE_C is disabled." );
-    }
 #endif /* MBEDTLS_CERTS_C */
+    }
 #endif /* MBEDTLS_CERTS_C && MBEDTLS_PEM_PARSE_C */
     if( ret != 0 )
     {
@@ -1209,8 +1209,8 @@ int main( int argc, char *argv[] )
         mbedtls_printf( "MBEDTLS_CERTS_C not defined." );
 #else
         mbedtls_printf( "All test keys loaded via MBEDTLS_CERTS_C are PEM-encoded, but MBEDTLS_PEM_PARSE_C is disabled." );
-    }
 #endif /* MBEDTLS_CERTS_C */
+    }
 #endif /* MBEDTLS_CERTS_C && MBEDTLS_PEM_PARSE_C */
     if( ret != 0 )
     {

--- a/programs/ssl/ssl_server2.c
+++ b/programs/ssl/ssl_server2.c
@@ -1658,8 +1658,8 @@ int main( int argc, char *argv[] )
         mbedtls_printf( "MBEDTLS_CERTS_C not defined." );
 #else
         mbedtls_printf( "All test CRTs loaded via MBEDTLS_CERTS_C are PEM-encoded, but MBEDTLS_PEM_PARSE_C is disabled." );
-    }
 #endif /* MBEDTLS_CERTS_C */
+    }
 #endif /* MBEDTLS_CERTS_C && MBEDTLS_PEM_PARSE_C */
     if( ret < 0 )
     {

